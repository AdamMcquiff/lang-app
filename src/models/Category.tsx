interface ICategoryModel {
    id?: number,
    title: string
}

export default ICategoryModel;