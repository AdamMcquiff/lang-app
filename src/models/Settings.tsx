export interface ISettingsModel {
    native_lang: string,
    translated_lang: string,
}