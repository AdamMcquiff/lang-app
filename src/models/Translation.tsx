interface ITranslationModel {
    id: number,
    native_word: string,
    translated_word: string,
    categoryId: number
}

export default ITranslationModel;