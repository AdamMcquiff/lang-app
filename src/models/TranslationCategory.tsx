interface ITranslationCategoryModel {
    id?: number,
    title: string
}

export default ITranslationCategoryModel;