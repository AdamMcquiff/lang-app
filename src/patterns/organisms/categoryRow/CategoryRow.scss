.m-category-row {
    font-family: $muli;
    color: $white;
    margin: { bottom: 2em; }
    
    display: grid;
    grid-template-columns: 1fr 3fr;
    
    
    header {
        border-right: 1px solid rgba($darker-blue, .5);
        margin: { right: 2em; }

        h2 {
            @include type-setting(2);
        }
    
        p {
            @include type-setting(0);
        }
    }

    & > div {
        display: grid;
        grid-template-columns: repeat(3, 1fr);
        grid-gap: .75em;

        @media screen and (min-width: 1240px) {
            grid-template-columns: repeat(4, 1fr);
        }
    }

    &__add-button {
        cursor: pointer;
        font-family: $muli;
        font-size: 1em;
        background-color: transparent;
        border: 2px dashed $lighter-blue;
        border-radius: $border-radius;
        box-shadow: $box-shadow;
        color: $white;
        min-height: 12.5em;
        transition: background-color .3s ease;

        &:hover,
        &:focus {
            background-color: rgba($lightest-blue, .5)
        }
    }
}