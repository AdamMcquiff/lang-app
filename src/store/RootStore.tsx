import SettingsService from '../services/settingsService/SettingsService';
import TranslationCategoryService from '../services/translationCategoriesService/TranslationCategoryService';
import TranslationService from '../services/translationService/TranslationService';
import SettingsStore from './SettingsStore';
import TranslationCategoryStore from './TranslationCategoryStore';
import TranslationStore from './TranslationStore';

class RootStore {
	public settingsStore: SettingsStore;
	public translationCategoryStore: TranslationCategoryStore;
	public translationStore: TranslationStore;

	constructor() {
		this.settingsStore = new SettingsStore(this, SettingsService);
		this.translationCategoryStore = new TranslationCategoryStore(this, TranslationCategoryService);
		this.translationStore = new TranslationStore(this, TranslationService);
	}
}

export default RootStore;