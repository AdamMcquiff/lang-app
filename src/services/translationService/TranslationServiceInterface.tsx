import { ITranslationModel } from '../../models/Translation';

export interface ITranslationService {
    list(): Promise<ITranslationModel[]>,
    create(translation: ITranslationModel): Promise<ITranslationModel>,
    update(translation: ITranslationModel): Promise<ITranslationModel>,
    delete(translation: ITranslationModel): Promise<ITranslationModel>
}