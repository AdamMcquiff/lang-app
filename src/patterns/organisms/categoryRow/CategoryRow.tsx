import { computed } from 'mobx';
import { observer } from 'mobx-react';
import * as React from 'react';
import ICategoryModel from '../../../models/Category';
import ITranslationModel from '../../../models/Translation';
import SettingsStore from '../../../store/SettingsStore';
import TranslationStore from '../../../store/TranslationsStore';
import Button from '../../atoms/button/Button';
import TranslationCard from '../../molecules/translationCard/TranslationCard';
import './CategoryRow.scss';

interface ICategoryProps {
    category: ICategoryModel,
    translationStore: TranslationStore,
    settingsStore: SettingsStore
}

@observer
class CategoryRow extends React.Component<ICategoryProps> {
    
    @computed get filteredTranslations() {
        return this.props.translationStore.translations.filter(
            (translation: ITranslationModel) => translation.categoryId === this.props.category.id
            )
        }
        
    constructor(props: ICategoryProps) {
        super(props);
        this.onAddTranslationButtonClick = this.onAddTranslationButtonClick.bind(this);
    }

    public onAddTranslationButtonClick() {
        return this.props.translationStore.createTranslation({
            categoryId: this.props.category.id, 
            native_word: '',
            translated_word: ''
        })
    }

    public render() {
        return (
            <div className="m-category-row">
                <h2>{this.props.category.title}</h2>

                {this.filteredTranslations.map((trans: ITranslationModel, id: number) => {
                    return <TranslationCard key={id}
                                            translation={trans}
                                            translationStore={this.props.translationStore}
                                            settingsStore={this.props.settingsStore} />
                })}
        
                <Button a11yLabel="Add new translation"
                        textLabel="Add new"
                        onClick={this.onAddTranslationButtonClick} />
            </div>
        );
    }
}

export default CategoryRow;
