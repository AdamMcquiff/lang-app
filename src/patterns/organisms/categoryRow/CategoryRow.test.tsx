import { mount, shallow } from 'enzyme';
import { when } from 'mobx';
import * as React from 'react';
import { ITranslationModel } from '../../../models/Translation';
import { ITranslationCategoryModel } from '../../../models/TranslationCategory';
import { ITranslationService } from '../../../services/translationService/TranslationServiceInterface';
import { RootStore } from '../../../store/RootStore';
import { CategoryRow } from './CategoryRow';

const translations = [{
    categoryId: 1,
    native_word: "Hello",
    translated_word: "Hallo"
}, {
    categoryId: 2,
    native_word: "Thanks",
    translated_word: "Danke"
}]

class MockTranslationService implements ITranslationService {
    public create(translation: ITranslationModel): Promise<ITranslationModel> {
        return Promise.resolve(translation);
    }

    public list(): Promise<ITranslationModel[]> {
        return Promise.resolve(translations);
    }

    public update(translation: ITranslationModel) {
        return Promise.resolve(translation);
    }

    public delete(translation: ITranslationModel) {
        return Promise.resolve(translation);
    }
}

const rootStore = new RootStore();
rootStore.translationStore.service = new MockTranslationService();

describe('CategoryRow', () => {
    const category: ITranslationCategoryModel = {
        id: 1,
        title: 'Test'
    };

    const props = {
        category,
        settingsStore: rootStore.settingsStore,
        translationStore: rootStore.translationStore
    };

    beforeEach(() => {
        rootStore.translationStore.list();
    });

    it('renders without crashing with all props', () => {
        const component = shallow(<CategoryRow {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('has label that correctly displays number of translations with 1 translation', () => {
        const component = mount(<CategoryRow {...props} />);
        const label = component.find('p').text();
        expect(label).toEqual("1 translation");
    });

    it('computed filteredTranslations method should contain one translation object', () => {
        const component = mount(<CategoryRow {...props} />);
        const instance = component.instance() as CategoryRow;

        expect(instance.filteredTranslations).toHaveLength(1);
        expect(instance.filteredTranslations).toEqual(
            expect.arrayContaining([
                expect.objectContaining({
                    categoryId: translations[0].categoryId,
                    native_word: translations[0].native_word,
                    translated_word: translations[0].translated_word,
                })
            ])
        );

        component.unmount();
    });

    it('computed totalNumberOfTranslations method should return a value of "1"', () => {
        const component = mount(<CategoryRow {...props} />);
        const instance = component.instance() as CategoryRow;

        expect(instance.totalNumberOfTranslations).toEqual(1);
        component.unmount();
    });

    it('should add a new translation on "Add" button click', (done) => {
        const component = mount(<CategoryRow {...props} />);
        const instance = component.instance() as CategoryRow;
        const onAddTranslationButtonClickMock = jest.spyOn<any, any>(instance, 'onAddTranslationButtonClick');

        expect(onAddTranslationButtonClickMock).toHaveBeenCalledTimes(0);
        expect(instance.totalNumberOfTranslations).toEqual(1);

        component
            .find('button.m-category-row__add-button')
            .simulate('click');
        
        when(() => instance.totalNumberOfTranslations === 2, () => {
            expect(instance.filteredTranslations).toHaveLength(2);
            done();
        });
    });
});