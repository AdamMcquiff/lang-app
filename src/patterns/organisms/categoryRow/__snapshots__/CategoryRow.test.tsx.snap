// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryRow renders without crashing with all props 1`] = `
<section
  className="m-category-row"
>
  <header>
    <h2>
      Test
    </h2>
    <p>
      1 translation
    </p>
  </header>
  <div>
    <TranslationCard
      key="0"
      settingsStore={
        SettingsStore {
          "rootStore": RootStore {
            "settingsStore": [Circular],
            "translationCategoryStore": TranslationCategoryStore {
              "create": [Function],
              "delete": [Function],
              "list": [Function],
              "rootStore": [Circular],
              "service": TranslationCategoryService {},
              "translationCategories": Array [],
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "TranslationCategoryStore@2.keys",
                  "observers": Set {},
                },
                "name": "TranslationCategoryStore@2",
                "target": [Circular],
                "values": Map {
                  "translationCategories" => Array [],
                },
              },
            },
            "translationStore": TranslationStore {
              "create": [Function],
              "delete": [Function],
              "list": [Function],
              "rootStore": [Circular],
              "service": MockTranslationService {},
              "translations": Array [
                Object {
                  "categoryId": 1,
                  "native_word": "Hello",
                  "translated_word": "Hallo",
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "TranslationStore@3.translations[..].keys",
                      "observers": Set {},
                    },
                    "name": "TranslationStore@3.translations[..]",
                    "pendingKeys": Map {
                      "cheerio" => false,
                      "$$typeof" => false,
                      "nodeType" => false,
                      "@@__IMMUTABLE_ITERABLE__@@" => false,
                      "@@__IMMUTABLE_RECORD__@@" => false,
                      "_isMockFunction" => false,
                      "toJSON" => false,
                    },
                    "proxy": [Circular],
                    "target": Object {
                      "categoryId": 1,
                      "native_word": "Hello",
                      "translated_word": "Hallo",
                      Symbol(mobx administration): [Circular],
                    },
                    "values": Map {
                      "categoryId" => 1,
                      "native_word" => "Hello",
                      "translated_word" => "Hallo",
                    },
                  },
                },
                Object {
                  "categoryId": 2,
                  "native_word": "Thanks",
                  "translated_word": "Danke",
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "TranslationStore@3.translations[..].keys",
                      "observers": Set {},
                    },
                    "name": "TranslationStore@3.translations[..]",
                    "pendingKeys": Map {
                      "cheerio" => false,
                      "$$typeof" => false,
                      "nodeType" => false,
                      "@@__IMMUTABLE_ITERABLE__@@" => false,
                      "@@__IMMUTABLE_RECORD__@@" => false,
                      "_isMockFunction" => false,
                      "toJSON" => false,
                    },
                    "proxy": [Circular],
                    "target": Object {
                      "categoryId": 2,
                      "native_word": "Thanks",
                      "translated_word": "Danke",
                      Symbol(mobx administration): [Circular],
                    },
                    "values": Map {
                      "categoryId" => 2,
                      "native_word" => "Thanks",
                      "translated_word" => "Danke",
                    },
                  },
                },
              ],
              "update": [Function],
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "TranslationStore@3.keys",
                  "observers": Set {},
                },
                "name": "TranslationStore@3",
                "target": [Circular],
                "values": Map {
                  "translations" => Array [
                    Object {
                      "categoryId": 1,
                      "native_word": "Hello",
                      "translated_word": "Hallo",
                      Symbol(mobx administration): ObservableObjectAdministration$$1 {
                        "defaultEnhancer": [Function],
                        "keysAtom": Atom$$1 {
                          "diffValue": 0,
                          "isBeingObserved": false,
                          "isPendingUnobservation": false,
                          "lastAccessedBy": 0,
                          "lowestObserverState": 2,
                          "name": "TranslationStore@3.translations[..].keys",
                          "observers": Set {},
                        },
                        "name": "TranslationStore@3.translations[..]",
                        "pendingKeys": Map {
                          "cheerio" => false,
                          "$$typeof" => false,
                          "nodeType" => false,
                          "@@__IMMUTABLE_ITERABLE__@@" => false,
                          "@@__IMMUTABLE_RECORD__@@" => false,
                          "_isMockFunction" => false,
                          "toJSON" => false,
                        },
                        "proxy": [Circular],
                        "target": Object {
                          "categoryId": 1,
                          "native_word": "Hello",
                          "translated_word": "Hallo",
                          Symbol(mobx administration): [Circular],
                        },
                        "values": Map {
                          "categoryId" => 1,
                          "native_word" => "Hello",
                          "translated_word" => "Hallo",
                        },
                      },
                    },
                    Object {
                      "categoryId": 2,
                      "native_word": "Thanks",
                      "translated_word": "Danke",
                      Symbol(mobx administration): ObservableObjectAdministration$$1 {
                        "defaultEnhancer": [Function],
                        "keysAtom": Atom$$1 {
                          "diffValue": 0,
                          "isBeingObserved": false,
                          "isPendingUnobservation": false,
                          "lastAccessedBy": 0,
                          "lowestObserverState": 2,
                          "name": "TranslationStore@3.translations[..].keys",
                          "observers": Set {},
                        },
                        "name": "TranslationStore@3.translations[..]",
                        "pendingKeys": Map {
                          "cheerio" => false,
                          "$$typeof" => false,
                          "nodeType" => false,
                          "@@__IMMUTABLE_ITERABLE__@@" => false,
                          "@@__IMMUTABLE_RECORD__@@" => false,
                          "_isMockFunction" => false,
                          "toJSON" => false,
                        },
                        "proxy": [Circular],
                        "target": Object {
                          "categoryId": 2,
                          "native_word": "Thanks",
                          "translated_word": "Danke",
                          Symbol(mobx administration): [Circular],
                        },
                        "values": Map {
                          "categoryId" => 2,
                          "native_word" => "Thanks",
                          "translated_word" => "Danke",
                        },
                      },
                    },
                  ],
                },
              },
            },
          },
          "service": SettingsService {},
          "settings": Object {
            "native_lang": "",
            "translated_lang": "",
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "SettingsStore@1.settings.keys",
                "observers": Set {},
              },
              "name": "SettingsStore@1.settings",
              "pendingKeys": Map {
                "cheerio" => false,
                "$$typeof" => false,
                "nodeType" => false,
                "@@__IMMUTABLE_ITERABLE__@@" => false,
                "@@__IMMUTABLE_RECORD__@@" => false,
                "_isMockFunction" => false,
                "toJSON" => false,
              },
              "proxy": [Circular],
              "target": Object {
                "native_lang": "",
                "translated_lang": "",
                Symbol(mobx administration): [Circular],
              },
              "values": Map {
                "native_lang" => "",
                "translated_lang" => "",
              },
            },
          },
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "SettingsStore@1.keys",
              "observers": Set {},
            },
            "name": "SettingsStore@1",
            "target": [Circular],
            "values": Map {
              "settings" => Object {
                "native_lang": "",
                "translated_lang": "",
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "SettingsStore@1.settings.keys",
                    "observers": Set {},
                  },
                  "name": "SettingsStore@1.settings",
                  "pendingKeys": Map {
                    "cheerio" => false,
                    "$$typeof" => false,
                    "nodeType" => false,
                    "@@__IMMUTABLE_ITERABLE__@@" => false,
                    "@@__IMMUTABLE_RECORD__@@" => false,
                    "_isMockFunction" => false,
                    "toJSON" => false,
                  },
                  "proxy": [Circular],
                  "target": Object {
                    "native_lang": "",
                    "translated_lang": "",
                    Symbol(mobx administration): [Circular],
                  },
                  "values": Map {
                    "native_lang" => "",
                    "translated_lang" => "",
                  },
                },
              },
            },
          },
        }
      }
      translation={
        Object {
          "categoryId": 1,
          "native_word": "Hello",
          "translated_word": "Hallo",
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "TranslationStore@3.translations[..].keys",
              "observers": Set {},
            },
            "name": "TranslationStore@3.translations[..]",
            "pendingKeys": Map {
              "cheerio" => false,
              "$$typeof" => false,
              "nodeType" => false,
              "@@__IMMUTABLE_ITERABLE__@@" => false,
              "@@__IMMUTABLE_RECORD__@@" => false,
              "_isMockFunction" => false,
              "toJSON" => false,
            },
            "proxy": [Circular],
            "target": Object {
              "categoryId": 1,
              "native_word": "Hello",
              "translated_word": "Hallo",
              Symbol(mobx administration): [Circular],
            },
            "values": Map {
              "categoryId" => 1,
              "native_word" => "Hello",
              "translated_word" => "Hallo",
            },
          },
        }
      }
      translationStore={
        TranslationStore {
          "create": [Function],
          "delete": [Function],
          "list": [Function],
          "rootStore": RootStore {
            "settingsStore": SettingsStore {
              "rootStore": [Circular],
              "service": SettingsService {},
              "settings": Object {
                "native_lang": "",
                "translated_lang": "",
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "SettingsStore@1.settings.keys",
                    "observers": Set {},
                  },
                  "name": "SettingsStore@1.settings",
                  "pendingKeys": Map {
                    "cheerio" => false,
                    "$$typeof" => false,
                    "nodeType" => false,
                    "@@__IMMUTABLE_ITERABLE__@@" => false,
                    "@@__IMMUTABLE_RECORD__@@" => false,
                    "_isMockFunction" => false,
                    "toJSON" => false,
                  },
                  "proxy": [Circular],
                  "target": Object {
                    "native_lang": "",
                    "translated_lang": "",
                    Symbol(mobx administration): [Circular],
                  },
                  "values": Map {
                    "native_lang" => "",
                    "translated_lang" => "",
                  },
                },
              },
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "SettingsStore@1.keys",
                  "observers": Set {},
                },
                "name": "SettingsStore@1",
                "target": [Circular],
                "values": Map {
                  "settings" => Object {
                    "native_lang": "",
                    "translated_lang": "",
                    Symbol(mobx administration): ObservableObjectAdministration$$1 {
                      "defaultEnhancer": [Function],
                      "keysAtom": Atom$$1 {
                        "diffValue": 0,
                        "isBeingObserved": false,
                        "isPendingUnobservation": false,
                        "lastAccessedBy": 0,
                        "lowestObserverState": 2,
                        "name": "SettingsStore@1.settings.keys",
                        "observers": Set {},
                      },
                      "name": "SettingsStore@1.settings",
                      "pendingKeys": Map {
                        "cheerio" => false,
                        "$$typeof" => false,
                        "nodeType" => false,
                        "@@__IMMUTABLE_ITERABLE__@@" => false,
                        "@@__IMMUTABLE_RECORD__@@" => false,
                        "_isMockFunction" => false,
                        "toJSON" => false,
                      },
                      "proxy": [Circular],
                      "target": Object {
                        "native_lang": "",
                        "translated_lang": "",
                        Symbol(mobx administration): [Circular],
                      },
                      "values": Map {
                        "native_lang" => "",
                        "translated_lang" => "",
                      },
                    },
                  },
                },
              },
            },
            "translationCategoryStore": TranslationCategoryStore {
              "create": [Function],
              "delete": [Function],
              "list": [Function],
              "rootStore": [Circular],
              "service": TranslationCategoryService {},
              "translationCategories": Array [],
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "TranslationCategoryStore@2.keys",
                  "observers": Set {},
                },
                "name": "TranslationCategoryStore@2",
                "target": [Circular],
                "values": Map {
                  "translationCategories" => Array [],
                },
              },
            },
            "translationStore": [Circular],
          },
          "service": MockTranslationService {},
          "translations": Array [
            Object {
              "categoryId": 1,
              "native_word": "Hello",
              "translated_word": "Hallo",
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "TranslationStore@3.translations[..].keys",
                  "observers": Set {},
                },
                "name": "TranslationStore@3.translations[..]",
                "pendingKeys": Map {
                  "cheerio" => false,
                  "$$typeof" => false,
                  "nodeType" => false,
                  "@@__IMMUTABLE_ITERABLE__@@" => false,
                  "@@__IMMUTABLE_RECORD__@@" => false,
                  "_isMockFunction" => false,
                  "toJSON" => false,
                },
                "proxy": [Circular],
                "target": Object {
                  "categoryId": 1,
                  "native_word": "Hello",
                  "translated_word": "Hallo",
                  Symbol(mobx administration): [Circular],
                },
                "values": Map {
                  "categoryId" => 1,
                  "native_word" => "Hello",
                  "translated_word" => "Hallo",
                },
              },
            },
            Object {
              "categoryId": 2,
              "native_word": "Thanks",
              "translated_word": "Danke",
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "TranslationStore@3.translations[..].keys",
                  "observers": Set {},
                },
                "name": "TranslationStore@3.translations[..]",
                "pendingKeys": Map {
                  "cheerio" => false,
                  "$$typeof" => false,
                  "nodeType" => false,
                  "@@__IMMUTABLE_ITERABLE__@@" => false,
                  "@@__IMMUTABLE_RECORD__@@" => false,
                  "_isMockFunction" => false,
                  "toJSON" => false,
                },
                "proxy": [Circular],
                "target": Object {
                  "categoryId": 2,
                  "native_word": "Thanks",
                  "translated_word": "Danke",
                  Symbol(mobx administration): [Circular],
                },
                "values": Map {
                  "categoryId" => 2,
                  "native_word" => "Thanks",
                  "translated_word" => "Danke",
                },
              },
            },
          ],
          "update": [Function],
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "TranslationStore@3.keys",
              "observers": Set {},
            },
            "name": "TranslationStore@3",
            "target": [Circular],
            "values": Map {
              "translations" => Array [
                Object {
                  "categoryId": 1,
                  "native_word": "Hello",
                  "translated_word": "Hallo",
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "TranslationStore@3.translations[..].keys",
                      "observers": Set {},
                    },
                    "name": "TranslationStore@3.translations[..]",
                    "pendingKeys": Map {
                      "cheerio" => false,
                      "$$typeof" => false,
                      "nodeType" => false,
                      "@@__IMMUTABLE_ITERABLE__@@" => false,
                      "@@__IMMUTABLE_RECORD__@@" => false,
                      "_isMockFunction" => false,
                      "toJSON" => false,
                    },
                    "proxy": [Circular],
                    "target": Object {
                      "categoryId": 1,
                      "native_word": "Hello",
                      "translated_word": "Hallo",
                      Symbol(mobx administration): [Circular],
                    },
                    "values": Map {
                      "categoryId" => 1,
                      "native_word" => "Hello",
                      "translated_word" => "Hallo",
                    },
                  },
                },
                Object {
                  "categoryId": 2,
                  "native_word": "Thanks",
                  "translated_word": "Danke",
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "TranslationStore@3.translations[..].keys",
                      "observers": Set {},
                    },
                    "name": "TranslationStore@3.translations[..]",
                    "pendingKeys": Map {
                      "cheerio" => false,
                      "$$typeof" => false,
                      "nodeType" => false,
                      "@@__IMMUTABLE_ITERABLE__@@" => false,
                      "@@__IMMUTABLE_RECORD__@@" => false,
                      "_isMockFunction" => false,
                      "toJSON" => false,
                    },
                    "proxy": [Circular],
                    "target": Object {
                      "categoryId": 2,
                      "native_word": "Thanks",
                      "translated_word": "Danke",
                      Symbol(mobx administration): [Circular],
                    },
                    "values": Map {
                      "categoryId" => 2,
                      "native_word" => "Thanks",
                      "translated_word" => "Danke",
                    },
                  },
                },
              ],
            },
          },
        }
      }
    />
    <Button
      a11yLabel="Add new translation"
      className="m-category-row__add-button"
      onClick={[Function]}
      textLabel="Add new"
    />
  </div>
</section>
`;
